src/charm.py

```diff
@@ -53,8 +54,19 @@ def _on_install(self, event) -> None:
         # Prevent the default cluster creation.
         self._cluster.inhibit_default_cluster_creation()
 
-        # Install the PostgreSQL packages.
-        self._install_apt_packages(event, ["postgresql"])
+        # Install the PostgreSQL and Patroni requirements packages.
+        self._install_apt_packages(event, ["postgresql", "python3-pip", "python3-psycopg2"])
```

Comment:
If this fails (due to a `subprocess.CalledProcessError` or `PackageNotFoundError`) the charm goes into `BlockedStatus` on lines [129](https://github.com/canonical/postgresql-operator/blob/6f0ef9eaaf039f56ff96ac6c3bd31a773f718e86/src/charm.py#L129_) and [137](https://github.com/canonical/postgresql-operator/blob/6f0ef9eaaf039f56ff96ac6c3bd31a773f718e86/src/charm.py#L137), but this isn't handled in the `install` hook, so I think it will go into the `BlockedStatus` and then try to execute the remaining code here. 

You'll want to add a check with a try or if statement. Up to you how you do it, but maybe:
```
# need to make _install_apt_packages return a boolean
install_successful = self._install_apt_packages(event, ["postgresql", "python3-pip", "python3-psycopg2"])
if not install_successful:
      return
```

Reply:
I'm interested in the conclusion of this discussion, since I don't have a strong opinion on how to handle blocking errors like this.

Reply:
Nice catch Mia! Thank you! I'll update this trying to get the best approach (also thinking about the on start hook that is part of the next PR of this group). Also, thanks for the suggestions on how to fix it.

Reply:
Code updated to correctly handle this situation. Thanks again, Mia. I'm using a try/except approach as [EAFP](https://docs.python.org/3.10/glossary.html#term-EAFP) seems to be the recommended way to do that.